packages/framework/src/client.ts

```diff
@@ -195,7 +195,7 @@ export class Client {
           throw new Error(`Invalid component: '${component}'`);
       }
     } else {
-      return result.data;
+      return result.data as T;
```

Comment:
`result.data` was incorrectly inferred as `any` due to lack of schema context, this is fixed as we now need to coerce to the validated type within a generic context.